### PR TITLE
Autoscaling test

### PR DESCRIPTION
**Pull request by yzdann**
### All Submissions:

* [ ] Have you signed our [CLA](https://www.mongodb.com/legal/contributor-agreement)?
* [ ] Put `closes #XXXX` in your comment to auto-close the issue that your PR fixes (if there is one).
* [ ] Update docs/release-notes/release-notes.md if your changes should be included in the release notes for the next release.

resolves https://github.com/commercetools/say_terabyte/issues/577

### CODE REVIEW

File: test/e2e/data/atlascluster_advanced_with_autoscaling.yaml [hunk 1]
```diff
@@ -0,0 +1,25 @@
+apiVersion: atlas.mongodb.com/v1
+kind: AtlasDeployment
+metadata:
+  name: atlascluster-basic
```
Comment (yzdann) [inline test/e2e/data/atlascluster_advanced_with_autoscaling.yaml:4]: atlascluster-adv

File: test/e2e/data/atlascluster_with_autoscaling.yaml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    providerSettings:
+      instanceSizeName: M30
+      providerName: AWS
+      regionName: US_EAST_1
+      autoScaling:
+        compute:
+          maxInstanceSize: M50
+          minInstanceSize: M30
+  processArgs:
+    javascriptEnabled: true
```
Comment (n3ph) [inline test/e2e/data/atlascluster_with_autoscaling.yaml:23]: No LF at EOF 😛

File: test/e2e/extra_test.go [hunk 3]
```diff
@@ -63,5 +63,19 @@ var _ = Describe("Deploy cluster", Label("cluster-extra-ns"), func() {
...
 			),
 		),
+		Entry("Advanced cluster configuration with no backup and no user", Label("ns-autoscaling"),
+			model.NewTestDataProvider(
+				"operator-ns-autoscaling",
+				model.AProject{},
+				model.NewEmptyAtlasKeyType().UseDefaulFullAccess(),
+				[]string{"data/atlascluster_with_autoscaling.yaml"},
+				[]string{},
+				[]model.DBUser{},
```
Comment (zonorti) [inline test/e2e/extra_test.go:73]: Do we need this user?

**Review by zonorti — APPROVED** [review]
LGTM, small nitpick

File: test/e2e/extra_test.go (hunk 3 above)
Comment (n3ph) [reply test/e2e/extra_test.go:73]: [Seems like](https://github.com/commercetools/mongodb-atlas-kubernetes/blob/8802b364bc593694b8c94fde51119a2c6d50d07f/test/e2e/model/dataprovider.go#L13)

**Review by n3ph — COMMENTED** [review]
Where is the test output? 😉

File: test/e2e/actions/extra_related.go [hunk 4]
```diff
@@ -40,3 +40,44 @@ func DeleteClusterCRD(data *model.TestDataProvider) {
 		})
 	})
 }
+
+func Autoscaling(data *model.TestDataProvider) {
+	// TODO: This is more integration test but because of ARM problem i need to run it as e2e for now
```
Comment (Freyert) [inline test/e2e/actions/extra_related.go:45]: Clean up extra comments.

File: test/e2e/extra_test.go [hunk 5]
```diff
@@ -63,5 +63,19 @@ var _ = Describe("Deploy cluster", Label("cluster-extra-ns"), func() {
 				},
 			),
 		),
+		Entry("Advanced cluster configuration with no backup and no user", Label("ns-autoscaling"),
```
Comment (Freyert) [inline test/e2e/extra_test.go:66]: I think the number one thing to work on is the narrative of the test so it's easier to follow.

I think this `Entry` should read something like `"Can manage advanced cluster autoscaling configuration"`.

Then your tests steps should have something like `"It won't scale down if `instanceSize` conflicts with autoscaled size"`. Potentially finding one of the tickets that refers to that issue and adding a comment.

**Review by Freyert — CHANGES_REQUESTED** [review]
Streamline the test narrative and clean the PR some. Awesome to have this in our tool belt again!

File: test/e2e/extra_test.go (hunk 3 above)
Comment (yzdann) [reply test/e2e/extra_test.go:73]: I answered this question here:
https://github.com/commercetools/mongodb-atlas-kubernetes/pull/1#discussion_r911784662

File: test/e2e/extra_test.go (hunk 5 above)
Comment (yzdann) [reply test/e2e/extra_test.go]: Thanks, i changed them please check it out again
https://github.com/commercetools/mongodb-atlas-kubernetes/pull/2/commits/ae6d00a71425b8bcf287290e67625b1e832f96b9

**Comment by yzdann** [comment]
> Please add a screenshot of the successful test result
@n3ph 
Screenshot 2022-07-14 at 12 17 55 PM

**Review by n3ph — APPROVED** [review]

**Review by Freyert — APPROVED** [review]

**Comment by zonorti** [comment]
@yzdann should we keep this PR open?

**Comment by yzdann** [comment]
@zonorti No, it's not necessary!

**Pull request closed without merging**